src/components/cc-logs/logs-controller.js

```diff
@@ -41,24 +43,9 @@ export class LogsController {
   }
 
   set filter (filter) {
-    const filterValuesByMetadataName = {};
-    if (Array.isArray(filter)) {
-      for (const predicate of filter) {
-        if (filterValuesByMetadataName[predicate.metadata] == null) {
-          filterValuesByMetadataName[predicate.metadata] = [];
-        }
-        filterValuesByMetadataName[predicate.metadata].push(predicate.value);
-      }
-    }
-    const filterValuesByMetadataNameEntries = Object.entries(filterValuesByMetadataName);
-
-    this._filterCallback = (log) => {
-      // NOTE: [].every() is always true, whatever the callback
-      return filterValuesByMetadataNameEntries.every(([metadata, values]) => {
-        const logMetadata = log.metadata.find((m) => m.name === metadata);
-        return values.includes(logMetadata?.value);
-      });
-    };
+    this._filterCallback = (filter == null)
+      ? truePredicate
+      : (log) => this._getMetadataFilterCallback(filter.metadata)(log) && this._getMessageFilterCallback(filter.message)(log);
```

Comment:
issue: it seems like we will call `_getMetadataFilterCallback` and `_getMessageFilterCallback` for each log, I'm not sure if it's a perf problem but I'd rather do:

```js
if (filter == null) {
  this._filterCallback = truePredicate;
}
else {
  const matchesMetadata = this._getMetadataFilterCallback(filter.metadata);
  const matchesMessage = this._getMessageFilterCallback(filter.message);
  this._filterCallback (log) => matchesMetadata(log) && matchesMessage(log);
}
```

Reply:
question: also, it seems like (not sure) checking the metadata is more costly so I would suggest to match on message first, WDYT?